examples/streaming-pwa/js/index.js

```diff
@@ -5,6 +5,24 @@ const searchField = document.querySelector("#search-field")
 const searchButton = document.querySelector("#search-button")
 const resultsContainer = document.querySelector("#results")
 
+// Quit if the browser doesn't support WritableStream
+if( window.WritableStream === undefined ){
```

Comment:
It maybe better to use the `typeof` check here:
`if (typeof window.ReadableStream === 'undefined' && typeof window.WritableStream === 'undefined')`

Reply:
Also accounting for Readable AND writable.

Reply:
I believe we can skip `typeof` here because we're getting `WritableStream` as property of the window object. the `typeof` check is useful because `typeof x` does not throw an error if x is undefined, while `x === undefined` throws a reference error. In this case, by referring to `WritableStream` as a property of `window` we can avoid this error, as trying to get an undefined property of an object just returns undefined e.g. `foo = {}; foo.bar === undefined // True`

We could absolutely write this as `typeof ReadableStream === "undefined" || typeof WritableStream === "undefined"`, and it's really just a matter of style. After looking at it, I personally prefer the typeof version.

The reason I only checked `WritableStream` is because no browser spec currently implements Writable without also implementing Readable (Firefox actually implements only Readable), but more verbosity won't hurt :)

In fact, checking `if (typeof window.ReadableStream === 'undefined' && typeof window.WritableStream === 'undefined')` would actually malfunction in firefox, due to the weirdness I mentioned.

Reply:
Man, thanks for the comment. I was definitely able to learn from it. Sounds good.